### PR TITLE
chore: fix type errors in bitfield-rle tests

### PR DESCRIPTION
**Pull request by EvanHahn**
This fixes all the type errors in the tests for `bitfield-rle`. Some of the type errors were entirely test-only, but others required changes to the "real" code. Most notably, many `Buffer` types were changed to `Uint8Array`.

### CODE REVIEW

File: src/core-manager/bitfield-rle.js [hunk 1]
```diff
@@ -31,13 +31,10 @@ class State {
 encode.bytes = 0
 
 /**
- * @param {Uint32Array} bitfield
- * @param {Buffer} [buffer]
- * @param {number} [offset]
+ * @param {import('type-fest').TypedArray} bitfield
+ * @returns {Buffer}
  */
-export function encode(bitfield, buffer, offset) {
```
Comment (EvanHahn) [inline src/core-manager/bitfield-rle.js:38]: These parameters were unused so I removed them.

File: src/core-manager/bitfield-rle.js [hunk 2]
```diff
@@ -31,13 +31,10 @@ class State {
 encode.bytes = 0
 
 /**
- * @param {Uint32Array} bitfield
- * @param {Buffer} [buffer]
- * @param {number} [offset]
+ * @param {import('type-fest').TypedArray} bitfield
```
Comment (gmaclennan) [inline src/core-manager/bitfield-rle.js:34]: Are you sure this is safe to do? I don't fully remember all of this code, but when writing it was specifically written for uint32Arrays - I think there are issues with endianess when accepting other options, given the modifications I made to the code.

**Review by gmaclennan — COMMENTED** [review]
I don't feel confident that this works correctly for typedarrays other than uint32Array. Do we need to make this change?

**Comment by EvanHahn** [comment]
I redid this to limit source changes.

`encodingLength` was only used in tests and its tests had tricky type errors. I made `encodingLength` private and removed the tests.

**Review by gmaclennan — APPROVED** [review]

**Pull request merged**